enhancements/update/duration-of-migration-to-multi-arch.md

```diff
@@ -0,0 +1,167 @@
+
+### Non-Goals
+
+Whether or not 'oc adm upgrade' is executed succesfully or the requested upgrade is accepted is not in the scope of this enhancement. So it is _not_ required to have test cases for this enhancement to cover (although they are valid in general):
+- upgrade from multi-arch to single-arch,
+- upgrade from multi-arch to multi-arch, or
+- upgrade a cluster in a cloud provider where multi-arch is not supported to multi-arch.
+
+The `oc adm upgrade status` might be useful to monitor the migration progress to multi-arch but the output of the command does not block this enhancement and its implementation. The `status` command is designed for cluster upgrades but migration is not an upgrade conceptually and is implemented through an upgrade.
+
+## Proposal
+
+The proposal includes the steps that an OpenShift cluster operator may take to provide information about the upgrade in additional to the existing names and version numbers for the components that the operator manages.
+
+* Add a few elements into the existing list `ClusterOperator.status.versions` in the `ClusterOperator` manifest. Those new `versions` contains the image pull specifications for the operator and OPTIONALLY the managed operands that the operator chooses to track. Although the version stays the same, the image pull specification are different if a cluster becomes multi-arch.
+
+* Populate the `ClusterOperator.status.versions` including these new elements above.
+
+The CVO has already implemented the way that it checks if each element in the `ClusterVersion.status.versions` from the payload's has the counterpart in the actual one on the cluster.
+
+There is no need to extend the APIs in `openshift/api`, or modify the CVO implementation.
+
+### Workflow Description
+
+This enhancement changes no workflow.
+
+### API Extensions
+
+None.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+[Hosted control planes enables the decoupling of updates between the control plane and the data plane](https://docs.openshift.com/container-platform/4.17/hosted_control_planes/hcp-updating.html#hcp-get-upgrade-versions_hcp-updating):
+
+* Update a control plane: specify `HostedCluster.spec.release`, and
+* Update a data plane: specify `NodePool.spec.release`,
+
+where the value is [the image pull specification of an OCP release payload image](https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1beta1.Release).
+
+Although hosted cluster migration to multi-arch is not triggered by "oc upgrade --multi-arch", `ClusterVersion.status` and `ClusterOperator.status` are still populated there the same way as they are on a standalone cluster. Thus, the proposal in this enhancement can be applied to a hosted cluster too.
+
+In addition, the following status can be exploited for the progress of an upgrade:
+
+* `HostedCluster.status.version` shows the status and `HostedCluster.status.payloadArch` indicates the arch.
```

Comment:
But [per the Godocs][1], `payloadArch` reflects the architecture of HostedCluster's `spec.release.image`, without provision for "and that target release has been rolled out enough to support cross-arch NodePools now".  So I don't think it's helpful for this duration-of-migration-to-multi-arch enhancement.

[1]: https://github.com/openshift/hypershift/blob/f8d1152772e6dce770b04784c5fcddf840f234bc/api/hypershift/v1beta1/hostedcluster_types.go#L1258

Reply:
I got it now from the Slack conversation.